.xo-config.js

```diff
@@ -44,7 +44,28 @@ module.exports = {
 		// Temporarily disabled
 		// TODO: Need to decide what to do with this rule. Either keep it disabled here or remove it from XO.
 		'@typescript-eslint/no-invalid-void-type': 'off', // Reason: https://github.com/sindresorhus/refined-github/pull/3168#discussion_r434479434
-		'unicorn/no-fn-reference-in-iterator': 'off'
+		'unicorn/no-fn-reference-in-iterator': 'off',
+		'@typescript-eslint/no-extra-non-null-assertion': 'error',
+		'@typescript-eslint/consistent-type-definitions':'error',
```

Comment:
While it might default to `interface`, this is not documented and also not clear.

Also, any reasoning to defaulting to `interface`?

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-definitions.md

The only benefit of `interface` is declaration merging, but `type` has other benefits too: https://stackoverflow.com/a/52682220

Reply:
> but `type` has other benefits too

The only benefit I see is that you can use primitives, like `type S = string`, but this rule doesn't seem to prevent us from using that.

From the replacements I see in this PR, it only affects `type`s that can as well be `interface`s.

In my eyes, the difference between the two:

- `type x = {};` looks like a declaration
- `interface x {}` looks like a definition

Minimal difference, really, it's just about consistency

Reply:
> The only benefit I see is that you can use primitives, like type S = string, but this rule doesn't seem to prevent us from using that.

Sure, but if you ever need that, you're back to inconsistency. Which is why I slightly prefer `type`, but it doesn't matter much. 🤷‍♂️

Kinda annoying that TS has such similar duplicate things.

Reply:
> any reasoning to defaulting to `interface`

I like definitions over declarations. The former looks cleaner to me:

```ts
interface X {}
type X = {};
```

Kinda like I prefer `class X {}` over `const X = class {}`